lib/Util.ts

```diff
@@ -268,21 +268,34 @@ class Util {
 	 */
 	public static merge(target: any, source: any) {
 		for (const key of Object.keys(source)) {
-			if (!target.hasOwnProperty(key) || typeof source[key] !== "object") {
-				// primitive/new value:
-				target[key] = source[key];
-			} else if (Array.isArray(source[key])) {
-				// skip array merge on target type mismatch:
-				if (!Array.isArray(target[key])) {
-					continue;
+			if (typeof source[key] === {} && Array.isArray(source[key])) {
```

Comment:
`typeof` returns a string, so compare with `"object"` or this won't pass ever (and it doesn't)
Then again this case might even be redundant as you're not meant to send arrays through merge again directly anyway.